### ISSUE TITLE
Win7 MinGW-w64 compile error: 'ssize_t'

### ISSUE THREAD

**Issue by ghost** [warmcat/libwebsockets#49]
when compiling the actual libwebsockets master branch with
MinGW-w64 (gcc 4.8.2) & CMake 2.8.12.2 on Win7 via the steps:

```
cmake -DWITH_SSL=0 -G "MinGW Makefiles" ..
mingw32-make.exe
```

it breaks at the make step with the error **conflicting types for 'ssize_t'**

```
D:\MinGW-w64\_home\libwebsockets-master\lib\libwebsockets.h:49:17:
    error:conflicting types for 'ssize_t'
    typedef SSIZE_T ssize_t;
In file included from d:\mingw-w64\mingw32\i686-w64-mingw32\include\crtdefs.h:10:0,
    from d:\mingw-w64\mingw32\i686-w64-mingw32\include\stdio.h:9,
    from D:\MinGW-w64\_home\libwebsockets-master\lib\base64-decode.c:42:
d:\mingw-w64\mingw32\i686-w64-mingw32\include\_mingw.h:389:13:
    note: previous declaration of 'ssize_t' was here
    typedef int ssize_t;
```

**Comment by kolen** [comment]
I still get this error in `v2.2.1` tag, [`v2.2.1` tarball](https://github.com/warmcat/libwebsockets/archive/v2.2.1.tar.gz) (latest release listed), but not in `v2.2-stable` tag or `master`.

```
[ 16%] Built target zlib_internal
[ 17%] Building C object CMakeFiles/websockets.dir/lib/base64-decode.c.obj
In file included from /tmp/libwebsockets-2.2.1/lib/private-libwebsockets.h:285:0,
                 from /tmp/libwebsockets-2.2.1/lib/base64-decode.c:43:
/tmp/libwebsockets-2.2.1/lib/libwebsockets.h:4343:17: error: conflicting types for ‘ssize_t’
 typedef SSIZE_T ssize_t;
                 ^~~~~~~
In file included from /usr/local/Cellar/mingw-w64/5.0.2_1/toolchain-i686/i686-w64-mingw32/include/stdio.h:9:0,
                 from /tmp/libwebsockets-2.2.1/lib/base64-decode.c:41:
/usr/local/Cellar/mingw-w64/5.0.2_1/toolchain-i686/i686-w64-mingw32/include/crtdefs.h:47:13: note: previous declaration of ‘ssize_t’ was here
 typedef int ssize_t;
             ^~~~~~~
make[2]: *** [CMakeFiles/websockets.dir/lib/base64-decode.c.obj] Error 1
make[1]: *** [CMakeFiles/websockets.dir/all] Error 2
make: *** [all] Error 2
```

Bisect shows that it's fixed in 2d6fd20c60e8f3b712f075ce88c599c1d3caa6ab.

**Comment by lws-team** [comment]
IIUI this boils down to, "please release a v2.2.2 tag", since everything already fixed on v2.2-stable... I have done that.
